### PR TITLE
[AIR][Telemetry] Record the entrypoint to AIR (`tune.run`, `Tuner.fit`, `Trainer.fit`, etc.)

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This PR adds telemetry for the entrypoint to Train/Tune by adding 1 usage tag: `AIR_ENTRYPOINT`

This tracks which method the user starts their training/tuning run with. This is a string that can be one of the following values:
1. `tune.run`
2. `Tuner.fit`
3. `Trainer.fit`
4. `tune.run_experiments`

This will allow us to make an educated decision about deprecating certain entrypoints in the future -- and only provide on canonical way of running AIR.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/air/_internal/usage.py [hunk 1]
```diff
@@ -280,3 +280,27 @@ def tag_ray_air_env_vars() -> bool:
...
+        run_experiments_api: Whether or not the user went through `tune.run_experiments`
+    """
+    if run_experiments_api:
+        tag = "tune.run_experiments"
+    elif not trainer_api and not tuner_api:
+        tag = "tune.run"
+    elif tuner_api and not trainer_api:
+        tag = "Tuner.fit"
+    elif trainer_api:
+        tag = "Trainer.fit"
```
Comment (gjoliver) [inline python/ray/air/_internal/usage.py:304]: what do you think if we don't add this util, and just record the raw flag values.
it's usually easier or more flexible to work with the raw data in data analysis.

File: python/ray/tune/tune.py [hunk 2]
```diff
@@ -340,6 +340,7 @@ def run(
     # a context object or similar.
     _tuner_api: bool = False,
     _trainer_api: bool = False,
+    _run_experiments_api: bool = False,
```
Comment (krfricke) [inline python/ray/tune/tune.py:343]: This becomes a bit much with the flags. It also looks like we can consolidate them as only certain combinations are possible:

```
Trainer.fit() --> _tuner_api=True, _trainer_api=True
Tuner.fit() --> _tuner_api=True, _trainer_api=False
tune.run() --> _tuner_api=False, _trainer_api=False
```

Can we just consolidate these into one (still internal) attribute `_api_entrypoint` with values `[Trainer.fit, Tuner.fit, tune.run, tune.run_experiments]`?

**Review by krfricke — COMMENTED** [review]
Generally LGTM but let's consolidate the flags

File: python/ray/air/_internal/usage.py (hunk 1 above)
Comment (justinvyu) [reply python/ray/air/_internal/usage.py:304]: Oh, you mean save the True/False flags? I think Kai's suggestion about consolidating all of these actually makes sense -- there are 3 sets of flags now. Would rather just save a single thing.

File: python/ray/tune/tune.py (hunk 2 above)
Comment (justinvyu) [reply python/ray/tune/tune.py]: Done! Made it an enum. I kept the TuneController stuff as `_trainer_api`, since all things there are "for train".

**Review by krfricke — APPROVED** [review]
Awesome, thanks!

**Review by matthewdeng — APPROVED** [review]

**Review by pcmoritz — APPROVED** [review]

**Review by gjoliver — APPROVED** [review]

**Pull request merged**